### PR TITLE
[Update] Make `vue/prop-name-casing` fixable

### PR DESCRIPTION
**Pull request by ota-meshi**
This PR makes `vue/prop-name-casing` fixable.
This implements autofix of rules proposed in #252

### CODE REVIEW

File: lib/rules/prop-name-casing.js [hunk 1]
```diff
@@ -46,7 +56,12 @@ function create (context) {
           data: {
             name: propName,
             caseType: caseType
-          }
+          },
+          fix: canFixPropertyName(item) ? fixer => {
+            return item.key.type === 'Literal'
+              ? fixer.replaceText(item.key, item.key.raw.replace(item.key.value, convertedName))
```
Comment (mysticatea) [inline lib/rules/prop-name-casing.js]: If the original name is `"🍻"`, what does it happen?

Comment (ota-meshi) [reply lib/rules/prop-name-casing.js]: Why!? It became "" !!!
Emoji is difficult...

**Comment by ota-meshi** [comment]
Thanks for review!
I changed it. to auto-fix if only for ASCII printable characters.

File: tests/lib/rules/prop-name-casing.js [hunk 2]
```diff
@@ -189,12 +217,156 @@ ruleTester.run('prop-name-casing', rule, {
...
+        export default {
+          props: {
+            [greeting_text]: String
+          }
+        }
+      `,
+      output: null,
+      parserOptions,
+      errors: [{
+        // bug ?
```
Comment (michalsnik) [inline tests/lib/rules/prop-name-casing.js]: I guess so, it shouldn't throw error in this case I think.

Comment (michalsnik) [reply tests/lib/rules/prop-name-casing.js]: But we can deal with it in separate task.

Comment (ota-meshi) [reply tests/lib/rules/prop-name-casing.js]: Thank you for your view
I wrote an issue #442

**Comment by F3n67u** [comment]
Make prop-name-casing fixable is very dangerous, because the fix code only fix the props name and leave the reference code same. I think the fix should leave to manual, we just show the alert and do nothing else.

**Pull request merged**